validate.d.ts

```diff
@@ -1,54 +1,323 @@
+    lessThanOrEqualTo?: number;
+
+    /** The input has to be less than this value. The error message is `must be less than %{count}` */
+    lessThan?: number;
+
+    /** The input has to be divisible by this value. The error message is `must be divisible by %{count}` */
+    divisibleBy?: number;
+
+    /** The input has to be odd. The error message is `must be odd` */
+    odd?: boolean;
+
+    /** The input has to be even. The error message is `must be even` */
+    even?: boolean;
+
+
+    /** Error message for and `strict` constraints. */
+    notValid?: string;
+
+    /** Error message for `onlyInteger` constraint */
+    notInteger?: string;
+
+    /** Error message for `greaterThan` constraint*/
+    notGreaterThan?: string;
+
+    /** Error message for `greaterThanOrEqualTo` constraint */
+    notGreaterThanOrEqualTo?: string;
+
+    /** Error message for `equalTo` constraint */
+    notEqualTo?: string;
+
+    /** Error message for `lessThan` constraint */
+    notLessThan?: string;
+
+    /** Error message for `lessThanOrEqualTo` constraint */
+    notLessThanOrEqualTo?: string;
+
+    /** Error message for `divisibleBy` constraint */
+    notDivisibleBy?: string;
+
+    /** Error message for `odd` constraint */
+    notOdd?: string;
+
+    /** Error message for `even` constraint */
+    notEven?: string;
+  };
+
+  /**
+   * The presence validator validates that the value is defined.
+   */
+  export type PresenceConstraint = boolean | {
+    /**
+     * Error message. Default is 'can't be blank'
+     */
+    message: string;
+
+    /**
+     * Set to false, to disallow the following values:
+     * - {} (empty objects)
+     * - [] (empty arrays)
+     * - "" (empty string)
+     * - " " (whitespace only string)
+     */
+    allowEmpty: boolean;
+  };
+    /**
+   * The URL validator ensures that the input is a valid URL.
+   */
+  export type UrlConstraint = boolean | {
+    /**
+     * A list of schemes to allow.
+     * If you want to support any scheme
+     * you can use a regexp here (for example [".+"]).
+     * The default value is ["http", "https"].
+     */
+    schemes?: string[];
+
+    /**
+     * The message if the validator fails.
+     * Defaults to is not a valid url
+     */
+    message: string;
+
+    /**
+     * A boolean that if true allows local hostnames
+     * such as 10.0.1.1 or localhost. The default is false.
+     */
+    allowLocal: boolean;
+  };
+
+export interface ConstraintsTypes {
+  date?: DateConstraint;
+
+  datetime?: DateConstraint;
+
+  email?: EmailConstraint;
+
+  equality?: EqualityConstraint;
+
+  exclusion?: ExclusionConstraint;
+
+  format?: FormatConstraint;
+
+  inclusion?: InclusionConstraint;
+
+  length?: LengthConstraint;
+
+  numericality?: NumericalityConstraint;
+
+  presence?: PresenceConstraint;
+
+  url?: UrlConstraint;
+
+}
+
+export interface Constraints {
+  [name: string]: ConstraintsTypes;
```

Comment:
NIT: Could we make use of `Record` here? `export type Constraints = Record<string, ConstraintsTypes>`